packages/cozy-scanner/src/ScannerQualification.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import { translate } from 'cozy-ui/transpiled/react/I18n'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+
+import DocumentQualification from './DocumentQualification'
+
+import { Modal as NewModal } from 'cozy-ui/transpiled/react'
+
+import { getTracker } from 'cozy-ui/transpiled/react/helpers/tracker'
+
+const pushAnalytics = qualification => {
+  const tracker = getTracker()
+  if (tracker) {
+    tracker.push(['trackEvent', 'Drive', 'Scanner', 'Add Qualification'])
+    if (qualification && qualification.label) {
+      tracker.push([
+        'trackEvent',
+        'Drive',
+        'Qualification',
+        qualification.label
+      ])
+    }
+  }
+}
+/**
+ * ScannerQualification component
+ *
+ * This component is used to be displayed just after taking a
+ * photo on a mobile device from the Scanner feature.
+ *
+ * It displays a modal with the `Qualification` Process.
+ * ATM the `Qualificiation` process is only about choosing the
+ * right category for a document but soon we'll have rename and
+ * one day OCR.
+ */
+class ScannerQualification extends Component {
+  state = {
+    qualification: undefined,
+    filename: ''
+  }
+
+  render() {
+    const { onSave, t, dismissAction } = this.props
+    const { qualification, filename } = this.state
+    return (
+      <NewModal
+        title={t('Scan.save_doc')}
+        dismissAction={dismissAction}
+        primaryText={t('Scan.save')}
+        primaryAction={async () => {
+          pushAnalytics(qualification)
+          try {
+            await onSave(qualification, filename)
+          } catch (error) {
+            console.error('error', error)
+            Alerter.error('Scan.error.generic')
+          }
+        }}
+        primaryType={'regular'}
+        secondaryText={t('Scan.cancel')}
+        secondaryAction={() => dismissAction()}
+        secondaryType={'secondary'}
+        description={
+          <DocumentQualification
+            onQualified={(qualification, filename) => {
+              this.setState({ qualification, filename })
+            }}
+            onFileNameChanged={filename => {
+              this.setState({ filename })
+            }}
+            editFileName={true}
+            title={t('Scan.qualify')}
+          />
+        }
+      />
+    )
+  }
+}
+
+ScannerQualification.propTypes = {
+  /**
+   * primary action of the modal
+   *
+   */
```

Comment:
Can you use comments like this `/** Primary action of the modal */` ?

It is a bit weird here since we do not know that there is a modal at this point.

Reply:
I think I don't get your point. 

We are in a component calling `ExperimentalModal`. We know that there is a modal at this point, no?

Reply:
Technically we're calling `ScannerQualification` when using these proptypes :)

Reply:
Yes for the user/reader, when reading the prop types, (s)he might not know that the implementation is through a modal. If the ScannerQualification component had `Modal` in its name, I think it would be clearer.